content/master/cli/command-reference.md

```diff
@@ -32,6 +32,135 @@ Client Version: v1.16.0
+Engine to pull and run composition functions. 
+
+{{<hint "important">}} 
+Running `crossplane render` requires [Docker](https://www.docker.com/).
+{{< /hint >}}
+
+Provide a composite resource, composition and composition function YAML 
+definition with the command to render the output locally. 
+
+For example, 
+`crossplane render xr.yaml composition.yaml function.yaml`
+
+The output includes the original composite resource followed by the generated 
+managed resources. 
+
+{{<expand "An example render output" >}}
+```yaml
+---
+apiVersion: nopexample.org/v1
+kind: XBucket
+metadata:
+  name: test-xrender
+status:
+  bucketRegion: us-east-2
+---
+apiVersion: s3.aws.upbound.io/v1beta1
+kind: Bucket
+metadata:
+  annotations:
+    crossplane.io/composition-resource-name: my-bucket
+  generateName: test-xrender-
+  labels:
+    crossplane.io/composite: test-xrender
+  ownerReferences:
+  - apiVersion: nopexample.org/v1
+    blockOwnerDeletion: true
+    controller: true
+    kind: XBucket
+    name: test-xrender
+    uid: ""
+spec:
+  forProvider:
+    region: us-east-2
+```
+{{< /expand >}}
+
+### Flags
+
+{{< table "table table-sm table-striped">}}
+| Short flag   | Long flag                             | Description                                           |
+| ------------ | -------------                         | ------------------------------                        |
+|              | `--context-files=<key>=<file>,<key>=<file>`    | A comma separated list of files to load for function "contexts." |
+|              | `--context-values=<key>=<value>,<key>=<value>` | A comma separated list of key-value pairs to load for function "contexts."                                                    |
+| `-r`         | `--include-function-results`          | Include the "results" or events from the function.   |
+| `-o`         | `--observed-resources=<directory or file>`               |
+Provide artificial managed resource data to the function.
+|
+| `-x`         | `--include-full-xr`          | Include a copy of the input Composite Resource spec and metadata fields in the rendered output.   |
+|              | `--timeout=`                          | Amount of time to wait for a function to finish.                    |
+{{< /table >}}
+
+The `crossplane render` command relies on standard 
+[Docker environmental variables](https://docs.docker.com/engine/reference/commandline/cli/#environment-variables) 
+to connect to the local Docker engine and run composition functions. 
+
+
+### Provide function context
+
+The `--context-files` and `--context-values` flags can provide data 
+to a function's `context`.  
+The context is JSON formatted data.
+
+### Include function results
+
+If a function produces Kubernetes events with statuses use the 
+`--include-function-results` to print them along with the managed resource 
+outputs. 
+
+### Include the composite resource 
+
+Composition functions can only change the `status` field of a composite 
```

Comment:
I'll do another full pass through but I think this limitation needs to be called out more specifically at a higher level.  It was common to use metadata.annotations and labels to store information, and there were examples in the docs showing this pattern, so breaking that pattern needs to be more clearly documented.   This also has impacts on the migration from legacy P&T to functions, since the existing compositions that patch into metadata need to be rewritten to use status instead.

Reply:
I definitely agree, but I'd rather raise a tracking issue for that and follow up ASAP rather than tackle it in this PR.

To get this PR finished as quickly as possible I'd like to keep it focused on the promotion to GA. I'm only adding new content where it pertains to the fact that functions are now GA, and native P&T is deprecated.